packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -526,23 +526,38 @@ export class DatabaseCluster extends DatabaseClusterNew {
    * Adds the single user rotation of the master password to this cluster.
    */
   public addRotationSingleUser(options: RotationSingleUserOptions = {}): secretsmanager.SecretRotation {
+    if (options.secret) {
+      //TODO: check given secret is not master secret of this!
+      //TODO: check given secret is not added already!
+      //TODO: how to generate proper ID?
+      const id = 'AdditionalSecretRotation';
+      return this.newSecretRotation(id, options);
+    } else {
+      return this.addRotationSingleUserMasterSecret(options);
+    }
+  }
+
+  private addRotationSingleUserMasterSecret(options: RotationSingleUserOptions): secretsmanager.SecretRotation {
```

Comment:
Name is a little awkward here. How about calling this `addSingleUserRotationForMasterSecret()`?